Makefile

```diff
@@ -14,7 +14,7 @@ $(if $(findstring $(space),$(TOPDIR)),$(error ERROR: The path to the OpenWrt dir
 
 world:
 
-DISTRO_PKG_CONFIG:=$(shell which -a pkg-config | grep -E '\/usr' | head -n 1)
+DISTRO_PKG_CONFIG:=$(shell command -v pkg-config)
```

Comment:
Is this correct?

Reply:
Probably not. Example:

```
~$ command -pv env
/bin/env
~$ command -v env
/usr/bin/env
~$ which -a env
/usr/bin/env
/bin/env
```

The original line gets all the entries and greps them to get the /usr one.

Maybe

```
DISTRO_PKG_CONFIG:=$(shell env PATH=/usr command -v pkg-config)
```

would work.

Reply:
btw the -p parameter for command should not be used. Only bash and maybe zsh support it.

Reply:
I mean should we just implement a local script to simulate the -a of which?

Reply:
Doing that just for this command seems overkill. The thread mentioned that this was a hack for Gentoo systems. Also, only one path is selected here.

Reply:
Looking at how angry that user was... we should really prefer the safe way.
But really that command just do what your command with PATH do... Don't really know...

Reply:
Oh funny. fish shell's command has a -a option that does the same thing.

Reply:
Anyway really a script that only 12 line would do the same thing....

Reply:
Let's ping @clefru just in case

Reply:
i mean really.. the script is just that o.o

```
#! /bin/sh

case $PATH in
	(*[!:]:) PATH="$PATH:" ;;
esac
IFS=:

for ELEMENT in $PATH; do
        PATH=$ELEMENT command -v "$@"
done
```

 ```
✘  ansuel@Ansuel-xps  ~/openwrt   master ± ./command_a.sh pkg-config
/usr/bin/pkg-config
/bin/pkg-config
 ✘  ansuel@Ansuel-xps  ~/openwrt   master ± ./command_a.sh env
/usr/bin/env
/bin/env
```

Reply:
We can consider adding some exit error or other stuff but really if THE ONLY TASK is find all occurrence of a command and parse what we want... then it's really this simple script...
@neheb should I change this pr to implement this?

Reply:
@neheb i implemented a simple sh to be on the safe side... What do you think about this?

Reply:
I don't know if it's really needed. for the pkgconfig case for example, path should just be set to /usr. It'd be interesting to see why DISTRO_PKG_CONFIG exists at all.

Reply:
@ynezz should have more info